cmd/helm/list.go

```diff
@@ -83,6 +83,27 @@ func newListCmd(cfg *action.Configuration, out io.Writer) *cobra.Command {
 			}
 
 			if client.Short {
+
+				names := make([]string, 0)
+				for _, res := range results {
+					//fmt.Fprintln(out, res.Name)
+					names = append(names, res.Name)
+				}
+
+				outputFlag := cmd.Flag("output")
+				if outputFlag.Changed {
+					switch outputFlag.Value.String() {
+					case "json":
+						output.EncodeJSON(out, names)
+						return nil
+					case "yaml":
+						output.EncodeYAML(out, names)
+						return nil
+					default:
+						return outfmt.Write(out, newReleaseListWriter(results))
+					}
+				}
+
 				for _, res := range results {
 					fmt.Fprintln(out, res.Name)
 				}
```

Comment:
If I use `helm list --short` the default value for the output is used of `table`. This shows a different result from the use of `helm list --short --output table` which is currently showing the same result as if it were not short.

I would suggest a few changes to this block...

1. Remove the `if outputFlag.Changed {` block. This section can format for all the types including the default.
2. Add a case for `table` (the default) and have its contents be the for loop currently listed after the if changed block.
3. Change the default case (not a known output format) to return an error of `fmt.Errorf("Unknown output format \"%s\"", outputFlag.Value.String())`. This is based on what's in Helm v2.

This would make the case statement handle all short forms in all cases.

Reply:
>If I use `helm list --short` the default value for the output is used of `table`. This shows a different result from the use of `helm list --short --output table` which is currently showing the same result as if it were not short.

Actually, I deployed Helm 2 and saw that there was no output format as `table` in helm.

>$ ./helm version
Client: &version.Version{SemVer:"v2.16.3", GitCommit:"1ee0254c86d4ed6887327dabed7aa7da29d7eb0d", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.16.3", GitCommit:"1ee0254c86d4ed6887327dabed7aa7da29d7eb0d", GitTreeState:"clean"
 $ ./helm list --short --output table
Error: Unknown output format "table

Hence thought that if the output is table then it should be the table, but ya  you are right, if default format is `table` then `helm list --short --output table` should also be the same as the output of `helm list --short`.

I will make the changes as given in your 1st and 2nd point.
Regarding the 3rd point. What I think is that in the `default` case of the `switch` if we return `outfmt.Write(out, newReleaseListWriter(results))` the if will throw the normal Helm 3 error message of bad format.
>╭─ansverma@localhost ~/contribution/helm/7749/helm2/helm ‹helm/7749/helm_list_short_listing_format*› 
╰─$ bin/helm list --short --output table
mysql
yoursql
╭─ansverma@localhost ~/contribution/helm/7749/helm2/helm ‹helm/7749/helm_list_short_listing_format*› 
╰─$ bin/helm list --short --output json 
["mysql","yoursql"]
╭─ansverma@localhost ~/contribution/helm/7749/helm2/helm ‹helm/7749/helm_list_short_listing_format*› 
╰─$ bin/helm list --short --output yaml
\- mysql
\- yoursql
╭─ansverma@localhost ~/contribution/helm/7749/helm2/helm ‹helm/7749/helm_list_short_listing_format*› 
╰─$ bin/helm list --short --output asdflkj
Error: invalid argument "asdflkj" for "-o, --output" flag: invalid format type

I think, the error message `Error: invalid argument "asdflkj" for "-o, --output" flag: invalid format type` is better than the one  in Helm 2.
I am currently making the changes. Please let me know if you want to have that particular error message which you mentioned anyways, I will do that.